packages/template-retail-react-app/app/hooks/use-basket.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCustomerBaskets, useCustomerId, useProducts} from 'commerce-sdk-react-preview'
+
+/**
+ * This hook combine some commerce-react-sdk hooks to provide more derived data for Retail App baskets
+ * @param id - basket id to get the current used basket among baskets returned, use first basket in the array if not defined
+ * @param shouldFetchProductDetail - boolean to indicate if the baskets should fetch product details based on basket items
+ */
+export const useBasket = ({id = '', shouldFetchProductDetail = false} = {}) => {
```

Comment:
The name of this newly-created hook is the same as `useBasket` in our commerce-sdk-react 🤔 

How about a different name to avoid confusion? Like `useActiveBasket` or `useCurrentBasket`.

Reply:
It includes information from the query result too. Maybe `useDerivedBasket`? 🤔

Reply:
That makes me think whether this hook returns too many things, which make us difficult to name the hook.

For example, from the way we currently use the hook, the outputs that we care about are mostly about the current active basket and some info about it. We don't use the plural `baskets`.

Reply:
That makes sense, I've made change to `useBasket` and just return some loading state with current active basket.